pynxtools/eln_mapper/schema_eln.py

```diff
@@ -0,0 +1,282 @@
+
+    Parameters
+    ----------
+    nxdl_element : ET.Element
+        This xml element that will be scanned through the descendants.
+    recursive_dict : Dict
+        A dict that store hierarchical structure of scheme eln.
+    root_name : str, optional
+        Name of root that user want to see to name their application, e.g. MPES,
+        by default 'ROOT_NAME'
+    reader_name : Prefered name of the reader.
+    is_root : bool, optional
+        Declar the elment as root or not, by default False
+    """
+
+    if is_root:
+        # Note for later: crate a new function to handle root part
+        nxdl = 'NX<NAME>.nxdl'
+        recursive_dict[root_name] = {'base_sections':
+                                     ['nomad.datamodel.metainfo.eln.NexusDataConverter',
+                                      'nomad.datamodel.data.EntryData']}
+
+        m_annotations: Dict = {'m_annotations': {'template': {'reader': reader_name,
+                                                              'nxdl': nxdl},
+                                                 'eln': {'hide': []}}}
+
+        recursive_dict[root_name].update(m_annotations)
+
+        recursive_dict = recursive_dict[root_name]
+
+    # Define quantities for taking care of field
+    quantities: Dict = None
+    subsections: Dict = None
+    for elm in nxdl_element:
+        tag = remove_namespace_from_tag(elm.tag)
+        # To skip NXentry group but only consider the child elments
+        if _should_skip_iteration(elm):
+            scan_xml_element_recursively(elm, recursive_dict)
+            continue
+        if tag == 'field':
+            if quantities is None:
+                recursive_dict['quantities'] = {}
+                quantities = recursive_dict['quantities']
+            construct_field_strucure(elm, quantities)
+        if tag == 'group':
+            if subsections is None:
+                recursive_dict['sub_sections'] = {}
+                subsections = recursive_dict['sub_sections']
+            construct_group_structure(elm, subsections)
+
+
+def get_eln_recursive_dict(recursive_dict: Dict, nexus_full_file: str) -> None:
+    """Develop a recursive dict that has hierarchical structure of scheme eln.
+
+    Parameters
+    ----------
+    recursive_dict : Dict
+        A dict that store hierarchical structure of scheme eln.
+    nexus_full_file : str
+        Full path of NeXus file e.g. <full_path>/paNXmpes.nxdl.xml
+    """
+
+    nxdl_root = ET.parse(nexus_full_file).getroot()
+    root_name = nxdl_root.attrib['name'][2:] if 'name' in nxdl_root.attrib else "<ROOT_NAME>"
+    recursive_dict['definitions'] = {'name': '<ADD PREFERED NAME>',
+                                     'sections': {}}
+    sections = recursive_dict['definitions']['sections']
+
+    scan_xml_element_recursively(nxdl_root, sections,
+                                 root_name=root_name, is_root=True)
+
+
+def generate_schema_eln(nexus_def: str, eln_file: str = None) -> None:
```

Comment:
eln_file_name I've seen that you name functions and variables already often good to indicate what is the intention behind these I think you can do this even more succinct, more a suggestions for the future